.asf.yaml

```diff
@@ -52,3 +52,24 @@ github:
     squash: true
     merge: false
     rebase: false
+
+  protected_branches:
+    master:
+      required_status_checks:
+        # strict means "Require branches to be up to date before merging".
+        strict: false
+        # contexts are the names of checks that must pass
+        contexts:
+          - Docker/build
+          - Frontend/build
+          - PR Lint/check
+          - E2E/Cypress
+          - lint
+          - test-mysql
+          - test-postgres
+          - test-sqlite
```

Comment:
It's a little unclear which names should the contexts use. 

This API returns only the job name, with Matrix suffix:

```
curl -s \
  -H "Authorization: token ${GITHUB_TOKEN}" \
  https://api.github.com/repos/apache/superset/commits/master/check-runs | jq ".check_runs[] | .name"

"test-sqlite (3.7)"
"pre-commit (3.7)"
"Prefer Typescript"
"test-postgres-presto (3.8)"
"test-postgres (3.8)"
"babel-extract (3.7)"
"test-postgres (3.7)"
"lint (3.7)"
"build"
"License Check"
"build"
"test-postgres-hive (3.8)"
"test-postgres-hive (3.7)"
"frontend-check"
"test-mysql (3.7)"
"build"
"babel-extract (3.7)"
```

Reply:
@ktmud do you have these actions set up in your fork? If so, can you check the name that appears under the settings tab for the repo?

Reply:
<img width="917" alt="status-check-names" src="https://user-images.githubusercontent.com/335541/105557505-46ba7380-5cc1-11eb-9195-38a912964f20.png">

I'm assuming the API will use the same names as displayed here?

We might want to rename a couple of our jobs to avoid confusion (there are a couple of duplicate "build"s).

I guess the safest route would be to have a very limited list here to override the admin settings, then add them back in a separate PR?

Reply:
I think those are the names we need to use, though they do look quite different than the names reported on PRs. I don't think the risk here is that high, I doubt committers will be merging PRs with failing status checks (even if they're not actually marked required). We could make and announcement just in case, "You might see some changes to the required checks as we figure things out, please don't merge anything that's failing a check"

Reply:
The docs show them as `<Action Name>/<Job Name>`: 
```       
 contexts:
          - gh-infra/jenkins
          - another/build-that-must-pass
   ```
although I realize that the action name was missing in some of the tests when I pulled them in, so I can update that. Looks to me that we would have to provide the action name in order for the duplicated job names to even work.

Reply:
I think the risk here is that if you add checks that do not exist or not matching the required context names as required, everyone will be blocked from merging until INFRA intervenes.

Reply:
Right, that's a good point. We should probably do this during hours of low activity. Worst case, we revert quickly and none of the check will be required. Without the admin repo privileges we are kind of shooting in the dark though

Reply:
Following on this scenario, do you think all the "blocked" prs would automatically be mergable after we revert this? Do you think there's any risk that we won't be able to unblock them? I think worst case scenario, we should be able to close and open the pr to retrigger the checks, which would just be time consuming.

Reply:
I think rather than revert we'd want to set the context as empty or remove the ones that end up stuck